src/net/udp.rs

```diff
@@ -169,27 +169,36 @@ impl UdpSocket {
     /// number of bytes written.
     pub async fn send_to<T: IoBuf>(
         &self,
-        buf: T,
+        buf: impl Into<Slice<T>>,
```

Comment:
The mutability contract is enforced through the `IoBuf`/`IoBufMut` bounds. For example, it's not possible to give a `&'static [u8]` to `read` because it does not implement `IoBufMut`. But I don't think the compiler is able to take this into account for dynamic buffers unless some deep inlining is possible; per the method signature, the input and return buffers may as well be different values altogether.